lib/chainer/dataset/convert.rb

```diff
@@ -30,13 +30,18 @@ def self.concat_examples(batch, device: nil, padding: nil)
       def self.concat_arrays(arrays, padding)
         unless arrays[0].kind_of?(Numo::NArray)
           arrays = Numo::NArray.cast(arrays)
+          if padding
+            return concat_arrays_with_padding(arrays, padding)
+          end
+          return arrays
         end
 
         if padding
           return concat_arrays_with_padding(arrays, padding)
         end
 
-        Numo::NArray.[](*arrays.to_a.map { |arr| arr.kind_of?(Numeric) ? arr : Numo::NArray.[](*arr) })
+        a = arrays.map{|arr| arr[:-, false]}
+        a[0].concatenate(*a[1..-1])
```

Comment:
this change is `Array` failed.

```ruby
arrays = [[1, 2], [3, 4]]

# before
> Numo::NArray.[](*arrays.to_a.map { |arr| arr.kind_of?(Numeric) ? arr : Numo::NArray.[](*arr) })
#=> Numo::Int32#shape=[2,2]
#[[1, 2], 
# [3, 4]]

# after
> a = arrays.map{|arr| arr[:-, false]}
# TypeError (no implicit conversion of Symbol into Integer)

Reply:
> this change is Array failed.
> ```
> arrays = [[1, 2], [3, 4]]
> ```

The expected array format is not "[[1, 2], [3, 4]]", 
"[Numo::Int32[1, 2], Numo::Int32[3, 4]]" is expected.

```
     def self.concat_arrays(arrays, padding)
        unless arrays[0].kind_of?(Numo::NArray)
          arrays = Numo::NArray.cast(arrays)
        end
```

In the above, it is requesting that Array[0] is NArray.

```
> arrays = [Numo::SFloat[1, 2], Numo::SFloat[3, 4]]
 => [Numo::SFloat#shape=[2]
[1, 2], Numo::SFloat#shape=[2]
[3, 4]] 
> Numo::NArray.[](*arrays.to_a.map { |arr| arr.kind_of?(Numeric) ? arr : Numo::NArray.[](*arr) })
 => Numo::DFloat#shape=[2,2]
[[1, 2], 
 [3, 4]] 
> a = arrays.map{|arr| arr[:-, false]}
 => [Numo::SFloat(view)#shape=[1,2]
[[1, 2]], Numo::SFloat(view)#shape=[1,2]
[[3, 4]]]
> a[0].concatenate(*a[1..-1])
 => Numo::SFloat#shape=[2,2]
[[1, 2], 
 [3, 4]] 
```

As you can see, It can process in the same class(SFloat).